velox/dwio/parquet/tests/reader/ParquetReaderCliTest.cpp

```diff
@@ -0,0 +1,251 @@
+              .withRowGroupSpecificData(kNumRowsPerGroup)
+              .build();
+
+      for (auto& batch : *batches) {
+        writer_->write(batch);
+      }
+      writer_->flush();
+      writer_->close();
+      files_.push_back(std::make_shared<LocalReadFile>(path));
+    }
+  }
+
+  std::shared_ptr<memory::MemoryPool> rootPool_;
+  std::shared_ptr<memory::MemoryPool> leafPool_;
+  std::shared_ptr<facebook::velox::exec::test::TempDirectoryPath>
+      fileDirectory_;
+  std::vector<std::shared_ptr<ReadFile>> files_;
+};
+
+TEST_F(ParquetReaderCliTest, readLocalFiles) {
+  auto rowType =
+      ROW({"bigint", "double", "string"}, {BIGINT(), DOUBLE(), VARCHAR()});
+  createFiles(3, rowType);
+
+  std::unique_ptr<FilterGenerator> filterGenerator =
+      std::make_unique<FilterGenerator>(rowType, 0);
+  auto filterBuilder = [&]() {
+    SubfieldFilters filters;
+    return filters;
+  };
+  auto scanSpec = filterGenerator->makeScanSpec(std::move(filterBuilder()));
+
+  for (auto file : files_) {
+    auto reader = ParquetReaderCli(
+        kDefaultReadingSize, rootPool_, leafPool_, file, rowType, scanSpec);
+    auto hasData = true;
+    uint32_t totalSize = 0;
+
+    while (hasData) {
+      auto result = BaseVector::create(rowType, 0, leafPool_.get());
+      auto numRowsRead = reader.read(result);
+
+      hasData = numRowsRead != 0;
+      if (hasData) {
+        auto rowVector = result->asUnchecked<RowVector>();
+
+        totalSize += rowVector->size();
+        EXPECT_EQ(numRowsRead, rowVector->size());
+      }
+    }
+
+    EXPECT_EQ(totalSize, kBatchCount * kNumRows);
+  }
+}
+
+TEST_F(ParquetReaderCliTest, readLocalFilesWithFilter) {
+  auto rowType =
+      ROW({"bigint", "double", "string"}, {BIGINT(), DOUBLE(), VARCHAR()});
+  createFiles(3, rowType);
+
+  std::unique_ptr<FilterGenerator> filterGenerator =
+      std::make_unique<FilterGenerator>(rowType, 0);
+  auto filterBuilder = [&]() {
+    SubfieldFilters filters;
+    filters[Subfield("bigint")] =
+        std::make_unique<facebook::velox::common::BigintRange>(
+            std::numeric_limits<int64_t>::min(),
+            std::numeric_limits<int64_t>::max(),
+            false);
+    return filters;
+  };
+  auto scanSpec = filterGenerator->makeScanSpec(std::move(filterBuilder()));
+
+  for (auto file : files_) {
+    auto reader = ParquetReaderCli(
+        kDefaultReadingSize, rootPool_, leafPool_, file, rowType, scanSpec);
+
+    auto hasData = true;
+    uint32_t totalSize = 0;
+
+    while (hasData) {
+      auto result = BaseVector::create(rowType, 0, leafPool_.get());
+      auto numRowsRead = reader.read(result);
+
+      hasData = numRowsRead != 0;
+
+      if (hasData) {
+        auto rowVector = result->asUnchecked<RowVector>();
+        totalSize += rowVector->size();
+      }
+    }
+
+    EXPECT_NE(totalSize, kBatchCount * kNumRows);
+  }
+}
+
+TEST_F(ParquetReaderCliTest, readLocalFilesWithSkip) {
+  auto rowType =
```

Comment:
I don't see much utility of `ParquetReaderCli`. Users still have to specify the file schema. There are examples for scan in the examples folder if users want to scan files.

Reply:
Hi Deepak, thanks for commenting. Before working on this PR, I have checked the examples and there is no Parquet related. This CLI can be used as a interface for users who only want to read Parquet files. And I also added a skip() feature. 

Would it be better to also have an example for the usages?

Reply:
Let's move the `skip()` API to another PR since that is its own discussion.
What Parquet related API are you looking for? I don't see any Parquet specific APIs in the current implementation.

Reply:
Sure. I can separate it into 2 PRs. 

The most important Parquet related API is read() (and skip() in the following PR). It is a streaming reading API. And after having this CLI, users can build the reader independently to read Parquet files.

Reply:
Can we extend https://github.com/facebookincubator/velox/blob/main/velox/examples/ScanOrc.cpp to achieve your needs?
We could improve it by inferring the file format using the `file` command instead of the user specifying it.

Reply:
Thanks for this suggestion. Our use case is to build the the Parquet Reader only as an independent component for users to read Parquet files efficiently on some specific offsets. My implementation was inspired by this example, but we only need the Parquet part to reduce the built library size.

Reply:
I see this as a helper util for all the readers. This should probably live in `dwio/common`.

Reply:
The term CLI is confusing since CLI tools are usually independent binaries. eg. parquet-cli from brew https://formulae.brew.sh/formula/parquet-cli